packages/xo-server/src/api/disk.mjs

```diff
@@ -182,33 +182,43 @@ async function handleImport(req, res, { type, name, description, vmdkData, srId,
             })()
           )
         } else {
+          let diskFormat = VDI_FORMAT_VHD
           await Promise.all(promises)
           part.length = part.byteCount
-          if (type === 'vmdk') {
-            vhdStream = await vmdkToVhd(part, vmdkData.grainLogicalAddressList, vmdkData.grainFileOffsetList)
-            size = vmdkData.capacity
-          } else if (type === 'vhd') {
-            vhdStream = part
-            const footer = await peekFooterFromVhdStream(vhdStream)
-            try {
-              checkFooter(footer)
-            } catch (e) {
-              if (e instanceof assert.AssertionError) {
-                throw new JsonRpcError(`Vhd file had an invalid header ${e}`)
+          switch (type) {
+            case 'vmdk':
+              vhdStream = await vmdkToVhd(part, vmdkData.grainLogicalAddressList, vmdkData.grainFileOffsetList)
+              size = vmdkData.capacity
+              break
+            case 'vhd':
+              {
+                const footer = await peekFooterFromVhdStream(vhdStream)
+                try {
+                  checkFooter(footer)
+                } catch (e) {
+                  if (e instanceof assert.AssertionError) {
+                    throw new JsonRpcError(`Vhd file had an invalid header ${e}`)
+                  }
+                }
+                vhdStream = part
+                size = footer.currentSize
               }
-            }
-            size = footer.currentSize
-          } else {
-            throw new JsonRpcError(`Unknown disk type, expected "vhd" or "vmdk", got ${type}`)
+              break
+            default:
```

Comment:
The type should be explicitly checked.

If you want to send `iso`, you can add this an alternate test for `raw`.